src/machine/machine_stm32_i2c_revb.go

```diff
@@ -86,8 +95,23 @@ func (i2c *I2C) Configure(config I2CConfig) error {
 
 // SetBaudRate sets the communication speed for I2C.
 func (i2c *I2C) SetBaudRate(br uint32) error {
```

Comment:
I'm not sure what the correct semantics we want are, but the logic here differs from NRF in that an unsupported value in NRF results in silently using 100KHz and will leave I2C enabled in any case.

NRF: https://github.com/tinygo-org/tinygo/blob/6384ecace093df2d0b93915886954abfc4ecfe01/src/machine/machine_nrf.go#L292

It seems like there's three possible behaviors for unsupported rates:
1. Change nothing (do validation before any changes) and return error
2. Force unsupported to the default 100KHz (hard to debug for by app devs?)
3. Disable the peripheral and return an error

Reply:
Updated to ensure that is what happens e.g. make no change and return error if invalid.